pkgs/applications/science/logic/bitwuzla/default.nix

```
@@ -0,0 +1,67 @@
+{ stdenv
+, fetchFromGitHub
+, fetchpatch
+, lib
+, python3
+, cmake
+, lingeling
+, btor2tools
+, gtest
+, gmp
+, cadical
+, minisat
+, picosat
+, cryptominisat
+, zlib
+, pkg-config
+}:
+
+stdenv.mkDerivation rec {
+  pname = "bitwuzla";
+  version = "2021-07-01";
+
+  src = fetchFromGitHub {
+    owner = "bitwuzla";
+    repo = "bitwuzla";
+    rev = "58d720598e359b1fdfec4a469c76f1d1f24db51a";
+    sha256 = "06ymqsdppyixb918161rmbgqvbnarj4nm4az88lkn3ri4gyimw04";
+  };
+
+  nativeBuildInputs = [ cmake pkg-config ];
+  buildInputs = [
+    lingeling
+    cadical
+    cryptominisat
+    picosat
+    minisat
+    btor2tools
+    gmp
+    zlib
+  ];
+
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS=ON"
+    "-DPicoSAT_INCLUDE_DIR=${lib.getDev picosat}/include/picosat"
+    "-DBtor2Tools_INCLUDE_DIR=${lib.getDev btor2tools}/include/btor2parser"
+    "-DBtor2Tools_LIBRARIES=${lib.getLib btor2tools}/lib/libbtor2parser.so"
+    "-DTESTING=YES"
+  ];
+
+  checkInputs = [ python3 gtest ];
+  doCheck = true;
+  preCheck = let
+    var = if stdenv.isDarwin then "DYLD_LIBRARY_PATH" else "LD_LIBRARY_PATH";
+  in
+    ''
+      export ${var}=$(readlink -f lib)
+      patchShebangs ..
+    '';
```

```suggestion
  preCheck = ''
    export ${if stdenv.isDarwin then "DYLD_LIBRARY_PATH" else "LD_LIBRARY_PATH"}=$(readlink -f lib)
    patchShebangs ..
  '';
```

Please be more specific about the patchShebangs.

---

a quick look says there is about one script per test so that looks unreasonable.

---

Is .. just the test directory or everything?

---

. is the build directory where cmake runs the build. .. is where the source is unpacked.

---

Can we specify the exact file(s) that need patching? Otherwise it takes a look at every file of the source and build a which is not very fast.

---

this discussion is going in circles.

---

Nothing is in circles here. patchShebangs points to the source checkout which also contains the build artifacts which means patchShebangs is slower. Just point it to the files that need patching. I am pretty sure the answer is some directory of files or a few files and not everything.